slurm/2_ApplicationSpecific/python/README.md

```diff
@@ -12,3 +12,27 @@ To run the Python script, simply submit the job to the scheduler from a login no
 ```
 $ sbatch python-sp.sh
 ```
+
+# Parallel Python Tutorial
+Parallel processing is a technique that executes multiple tasks at the same time using multiple CPU cores. This directory includes examples of two ways to perform parallel processing in Python.
+
+## Multiprocessing ([fibonacci_joblib.py](./fibonacci_multiproc.py))
+There are numerous APIs available to run python code in parallel, each with their strengths and weaknesses. A common API for parallel python processing is called `multiprocessing`. This library is powerful, enabling deep functionality like interprocess communication. However, for this simple demo we will stick to a very basic example.
+
+The `fibonacci_multiproc.py` script demonstrates using a process pool to parallelized computations. The `with with Pool(n_jobs) as p:` line creates a pool of `n_jobs` number of processes which can then execute code in parallel. The `p.map(fib, my_values)` line then applies the `fib` function from the serial example to a list of integers called `my_values`. The `multiprocessing` library then handles all of the process management for you as computation is run in parallel. The `multiprocessing` API provides many tools to handle process management beyond this simple example, you can find more information on all of these functions in [Python's documentation](https://docs.python.org/3/library/multiprocessing.html). 
+
+Please note, we specify the number of parallel processes with the `n_jobs` variable in our `for` loop. The value you select for the number of parallel processes should match the number of CPUs or tasks you request for your job in order to see runtime improvements in your parallel processes as Python cannot run multiple processes on a single CPU. Furthermore, there is overhead when creating and managing each process, so arbitrarily increasing `n_jobs` may not always yield faster program runtimes.
+
+## Joblib ([fibonacci_joblib.py](./fibonacci_joblib.py))
+For tasks that are embarrassingly parallel or those using NumPy arrays, `joblib` can be a more efficient and convenient solution. Since our `multiprocessing` example above involves computing fibonacci numbers in separate processes without any dependencies across processes, this computation is considered **embarassingly parallel**.  Thus, we can use `joblib` to compute Fibonacci numbers in parallel.
```

Comment:
```suggestion
For tasks that are [embarrassingly parallel](https://en.wikipedia.org/wiki/Embarrassingly_parallel) or those using NumPy arrays, `joblib` can be a more efficient and convenient solution. Since our `multiprocessing` example above involves computing fibonacci numbers in separate processes without any dependencies across processes, this computation is considered **embarassingly parallel**.  Thus, we can use `joblib` to compute Fibonacci numbers in parallel.
```
Suggest we include a link to the Wikipedia page describing embarrassingly parallel